lightning/src/ln/channelmanager.rs

```diff
@@ -8816,8 +8816,15 @@ where
 			None => builder,
 			Some(payer_note) => builder.payer_note(payer_note),
 		};
+
 		let invoice_request = builder.build_and_sign()?;
 
+		let total_liquidity: u64 = self.list_usable_channels().iter().map(|channel| channel.next_outbound_htlc_limit_msat).sum();
+		if invoice_request.amount_msats() > Some(total_liquidity) {
```

Comment:
This will fail in the case where we have an offer with a value denominated in bitcoin and caller doesn't provide an amount. We should use the value in the offer in that case.

Reply:
Got it updated!